rosidl_generator_py/rosidl_generator_py/__init__.py

```diff
@@ -0,0 +1,146 @@
+        subfolder = os.path.basename(os.path.dirname(ros_interface_file))
+        import_path = convert_path_to_import(os.path.join(base_path, subfolder))
+        if extension == '.msg':
+            spec = parse_message_file(args['package_name'], ros_interface_file)
+            mapping = mapping_msgs
+            type_name = spec.base_type.type
+        elif extension == '.srv':
+            spec = parse_service_file(args['package_name'], ros_interface_file)
+            mapping = mapping_srvs
+            type_name = spec.srv_name
+        else:
+            continue
+
+        modules[subfolder].append(type_name)
+        for template_file, generated_filename in mapping.items():
+            data = {'import_path': import_path, 'spec': spec, 'subfolder': subfolder}
+            data.update(functions)
+            generated_file = os.path.join(
+                args['output_dir'], subfolder, generated_filename % type_name)
+            expand_template(
+                template_file, data, generated_file,
+                minimum_timestamp=latest_target_timestamp)
+
+    open(os.path.join(args['output_dir'], '__init__.py'), 'w').close()
+    for module in modules:
+        with open(os.path.join(args['output_dir'], module, '__init__.py'), 'w') as f:
+            for type_ in modules[module]:
+                f.write('from %s._%s import %s\n' % (import_path, type_, type_))
+
+    return 0
+
+
+def convert_path_to_import(path):
+    # Based on:
+    # https://www.safaribooksonline.com/library/view/python-cookbook/0596001673/ch04s16.html
+    allparts = []
+    while 1:
+        parts = os.path.split(path)
+        if parts[0] == path:  # sentinel for absolute paths
+            allparts.insert(0, parts[0])
+            break
+        elif parts[1] == path:  # sentinel for relative paths
+            allparts.insert(0, parts[1])
+            break
+        else:
+            path = parts[0]
+            allparts.insert(0, parts[1])
+    return '.'.join(allparts)
+
+
+def value_to_py(type_, value):
+    assert type_.is_primitive_type()
+    assert value is not None
+
+    if not type_.is_array:
+        return primitive_value_to_py(type_, value)
+
+    py_values = []
+    for single_value in value:
+        py_value = primitive_value_to_py(type_, single_value)
+        py_values.append(py_value)
+    py_value = '[%s]' % ', '.join(py_values)
+    return py_value
+
+
+def primitive_value_to_py(type_, value):
+    assert type_.is_primitive_type()
+    assert value is not None
+
+    if type_.type == 'bool':
+        return 'True' if value else 'False'
+
+    if type_.type in [
+        'byte',
+        'char',
+        'int8', 'uint8',
+        'int16', 'uint16',
+        'int32', 'uint32',
+        'int64', 'uint64',
+    ]:
+        return str(value)
+
+    if type_.type in ['float32', 'float64']:
+        return '%s' % value
+
+    if type_.type == 'string':
+        return '"%s"' % escape_string(value)
```

Comment:
Can we use single quotes around the string here? And escape the single quotes inside.